code/game/atoms.dm

```diff
@@ -228,22 +228,22 @@
 /atom/proc/in_contents_of(container)//can take class or object instance as argument
 	if(ispath(container))
 		if(istype(src.loc, container))
-			return 1
+			return TRUE
 	else if(src in container)
-		return 1
+		return TRUE
+	return FALSE
+
+/atom/proc/get_examine_name(mob/user)
+	var/list/inject_before = list()
+	var/list/inject_after = list()
+	SendSignal(COMSIG_ATOM_GET_EXAMINE_NAME, user, inject_before, inject_after)
+	. = "[gender == PLURAL? "some" : "a"] [length(inject_before)? "[english_list(inject_before, "")] " : ""][src][length(inject_after)? " [english_list(inject_after, "")]" : ""]"

```

Comment:
You don't need these before/after lists and what if a component wants to modify the name itself and hide it?
```
. = "\a [src]"
var/list/f_name_list = list("\a", " ", "[src]")
if(SendSignal(COMSIG_ATOM_F_NAME, f_name_list) & F_NAME_CHANGED)
  . = f_name_list.Join("")
```